### ISSUE TITLE
Dual GPU bug

### ISSUE THREAD

**Issue by Xyndra** [FoundryMC/Veil#17]
```
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]: Dear ImGui Assertion Failed: (g.CurrentWindowStack.Size == 1) && "Mismatched Begin/BeginChild vs End/EndChild calls: did you forget to call End/EndChild?"
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]: Assertion Located At: /tmp/imgui/jni/imgui.cpp:8157
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]: java.lang.Exception: Stack trace
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at java.base/java.lang.Thread.dumpStack(Thread.java:1391)
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at imgui.ImGui$1.imAssertCallback(ImGui.java:94)
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at imgui.assertion.ImAssertCallback.imAssert(ImAssertCallback.java:21)
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at imgui.ImGui.endFrame(Native Method)
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at foundry.veil.impl.client.imgui.VeilImGuiImpl.begin(VeilImGuiImpl.java:53)
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at foundry.veil.api.client.render.VeilRenderSystem.beginFrame(VeilRenderSystem.java:643)
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at net.minecraft.client.Minecraft.handler$zdk000$veil$beginFrame(Minecraft.java:3502)
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at net.minecraft.client.Minecraft.runTick(Minecraft.java:1210)
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at net.minecraft.client.Minecraft.updateScreenAndTick(Minecraft.java:2210)
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at net.minecraft.client.Minecraft.clearLevel(Minecraft.java:2181)
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at net.minecraft.client.Minecraft.emergencySave(Minecraft.java:1392)
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at net.minecraft.client.Minecraft.run(Minecraft.java:828)
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at net.minecraft.client.main.Main.main(Main.java:250)
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
[21:16:58] [Render thread/INFO] (Minecraft) [STDERR]:   at net.fabricmc.devlaunchinjector.Main.main(Main.java:86)
```
The game crashed when I opened the debug thing and clicked my secondary GPU on device info

**Comment by Ocelot5836** [comment]
Can you show the log from furthur up?

**Comment by Xyndra** [comment]
```
[10:44:30] [Render thread/ERROR] (Minecraft) Unreported exception thrown!
 java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at foundry.veil.impl.client.editor.DeviceInfoViewer.renderOpenCL(DeviceInfoViewer.java:203) ~[Veil-fabric-1.20.1-1.0.0.258.jar:1.0.0.258]
	at foundry.veil.impl.client.editor.DeviceInfoViewer.renderComponents(DeviceInfoViewer.java:262) ~[Veil-fabric-1.20.1-1.0.0.258.jar:1.0.0.258]
	at foundry.veil.api.client.editor.SingleWindowEditor.render(SingleWindowEditor.java:20) ~[Veil-fabric-1.20.1-1.0.0.258.jar:1.0.0.258]
	at foundry.veil.impl.client.editor.DeviceInfoViewer.render(DeviceInfoViewer.java:284) ~[Veil-fabric-1.20.1-1.0.0.258.jar:1.0.0.258]
	at foundry.veil.api.client.editor.EditorManager.render(EditorManager.java:116) ~[Veil-fabric-1.20.1-1.0.0.258.jar:1.0.0.258]
	at foundry.veil.impl.client.imgui.VeilImGuiImpl.begin(VeilImGuiImpl.java:59) ~[Veil-fabric-1.20.1-1.0.0.258.jar:1.0.0.258]
	at foundry.veil.api.client.render.VeilRenderSystem.beginFrame(VeilRenderSystem.java:643) ~[Veil-fabric-1.20.1-1.0.0.258.jar:1.0.0.258]
	at net.minecraft.client.Minecraft.handler$zkp000$veil$beginFrame(Minecraft.java:5002) ~[minecraft-merged-b313045ff1-1.20.1-loom.mappings.1_20_1.layered+hash.2198-v2.jar:?]
	at net.minecraft.client.Minecraft.runTick(Minecraft.java:1210) ~[minecraft-merged-b313045ff1-1.20.1-loom.mappings.1_20_1.layered+hash.2198-v2.jar:?]
	at net.minecraft.client.Minecraft.run(Minecraft.java:802) ~[minecraft-merged-b313045ff1-1.20.1-loom.mappings.1_20_1.layered+hash.2198-v2.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:250) ~[minecraft-merged-b313045ff1-1.20.1-loom.mappings.1_20_1.layered+hash.2198-v2.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86) ~[dev-launch-injector-0.2.1+build.8.jar:?]
```
